README.md

```diff
@@ -315,3 +315,54 @@ clconf \
     getv / \
     --template-string '{{ cgetv "/db/username" }}:{{ cgetv "/db/password" }}'
 ```
+
+### Templating
+
+`clconf` has a `template` operation that functions as a
+[confd](https://github.com/kelseyhightower/confd) replacement but supports only
+yaml as a value store. It uses command line arguments in place of `confd`'s
+[toml files](https://github.com/kelseyhightower/confd/blob/master/docs/template-resources.md)
+to determine where templates are found and output placed.
+
+All of the options for `getv` are available for specifying yaml sources,
+and the templates behave as outlined above. The `template` operation takes it
+a step further by templating many files in a single run. The `template`
+function's `--help` provides examples:
+
+```bash
+This will take an arbitrary number of source templates (or folders full
+of templates) and process them either in place (see --in-place) or into the
+folder specified as the last argument. It will make any folders required
+along the way. If a source is an existing file (not a folder) it will be
+treated as a template regardless of the extension (though if the extension
+matches it will still be removed).
+
+Usage:
+  clconf template <src1> [src2...] [destination folder] [flags]
+
+Examples:
+  # Apply all templates with the .clconf extension to their relative folders in /dest
+  template /tmp/srcFolder1 /tmp/srcFolder2 /dest
+
+  # Apply all templates in both folders with the .clconf extension to the root of /dest
+  template /tmp/srcFolder1 /tmp/srcFolder2 /dest --flatten
+
+  # Interpret /tmp/srcFile.sh where it is (result is /tmp/srcFile.sh)
+  template /tmp/srcFile.sh --in-place
+
+  # Interpret /tmp/srcFile.sh.clconf where it is (result is /tmp/srcFile.sh)
+  template /tmp/srcFile.sh.clconf --in-place
+
+  # Interpret /tmp/srcFile.sh.clconf where it is (result is /tmp/srcFile.sh.clconf)
+  template /tmp/srcFile.sh.clconf --in-place --template-extension ""
+
+Flags:
+      --dir-mode string             Chmod mode (e.g. 755) to apply to newly created directories. (default "775")
+      --file-mode string            Chmod mode (e.g. 644) to apply to files when templating (new and existing) (defaults to copy from source template).
+      --flatten                     Don't preserve relative folders when processing a source folder.
+  -h, --help                        help for template
+      --in-place                    Template the files in the folder they're found (implies no destination)
+      --keep-existing-permissions   Only apply --file-mode to new files, leave existing files as-is.
+      --rm                          Remove template files after processing.
+      --template-extension string   Template file extension (will be removed during templating). (default ".clconf")
+```
```

Comment:
I think this is a bad idea.  It will almost certainly become outdated at the next modification, _and_ is already available as `--help`.  I think we should remove this and just say something like:

> For detailed usage see `clconf template --help`.

and leave it at that.

Reply:
So I went back and forth on this. I initially started writing complex examples in the doc, but that leaves a couple problems:

1. If we had different examples here, why wouldn't we put them in the command itself?
1. The biggest risk to docs like this is that they eventually lie, if we put the output of the command in here, it's clear that the command itself is the real source of docs and you should look there if something doesn't match
1. When the command inevitably expand, keeping the docs updated is a simple as copying the result of `--help` back in here. No combining through each statement and ensuring it's still accurate

It felt weird to paste this in there, but I couldn't come up with a reason _why_ it felt weird. If you have one I'd definitely like to hear it.

And if we just do the `see <blah>` for complete usage it leaves the user not really knowing what the tool _does_ by reading the README, which I feel is the whole goal of the README.